bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,291 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.handler;
+
+import static java.util.Collections.emptySortedSet;
+import static java.util.Objects.requireNonNull;
+import static java.util.Objects.requireNonNullElse;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.openhab.core.thing.ThingStatus.OFFLINE;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @NonNullByDefault({})
+    private String username;
+    @NonNullByDefault({})
+    private char[] password;
+    @NonNullByDefault({})
+    private String url;
+    private long refreshInterval;
+    private long propertiesRefreshInterval;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = requireNonNull(httpClientFactory, "httpClientFactory");
```

Comment:
httpClientFactory can never be null here
```suggestion
        this.httpClientFactory = httpClientFactory;
```